scripts/earthengine/events_pipeline.py

```diff
@@ -0,0 +1,818 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script to generate events from source data.
+It is a pipeline with the following steps:
+  1. get_source_data: Download data from source, incrementally with any source specific processing
+  2. generate_gridded_csv: Process source data into CSV with S2 cells.
+  3. generate_events: Generate events and related stats from the csv data.
+"""
```

Comment:
nit: since this is the most crucial part of the processing, I'd love to have more comments. For example:

is there an order to the processing? is it done in parallel? what if there is a failure in some stage of the pipeline (does it need to be re-run or can it just resume from where it left of)? etc.

Reply:
+1

Reply:
Added more comments on these points.